pkg/server/middleware.go

```diff
@@ -8,28 +8,38 @@ import (
 	"strings"
 
 	"github.com/openshift/console/pkg/auth"
+	"github.com/openshift/console/pkg/serverutils"
 
 	"k8s.io/klog"
 )
 
 // Middleware generates a middleware wrapper for request hanlders.
 // Responds with 401 for requests with missing/invalid/incomplete token with verified email address.
-func authMiddleware(a *auth.Authenticator, hdlr http.HandlerFunc) http.Handler {
+func authMiddleware(authers map[string]*auth.Authenticator, hdlr http.HandlerFunc) http.Handler {
 	f := func(user *auth.User, w http.ResponseWriter, r *http.Request) {
 		hdlr.ServeHTTP(w, r)
 	}
-	return authMiddlewareWithUser(a, f)
+	return authMiddlewareWithUser(authers, f)
 }
 
-func authMiddlewareWithUser(a *auth.Authenticator, handlerFunc func(user *auth.User, w http.ResponseWriter, r *http.Request)) http.Handler {
+func authMiddlewareWithUser(authers map[string]*auth.Authenticator, handlerFunc func(user *auth.User, w http.ResponseWriter, r *http.Request)) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		user, err := a.Authenticate(r)
+		// Get the correct Auther for the cluster.
+		cluster := serverutils.GetCluster(r)
+		auther, autherFound := authers[cluster]
+
+		if !autherFound {
+			klog.Errorf("Bad Request. Invalid cluster: %v", cluster)
+			w.WriteHeader(http.StatusBadRequest)
```

Comment:
We should make sure the error reason is returned to the client in the response, otherwise this will be tough to debug.